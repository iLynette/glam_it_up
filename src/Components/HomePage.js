import { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchProductFromApi } from '../redux/Products/products';
import Categories from './Categories';
import Btn from './Arrow';

const HomePage = () => {
  const data = useSelector((state) => state.productReducer);
  const [value, setValue] = useState('');
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchProductFromApi());
  }, []);

  const handleClick = (e) => {
    setValue(e.target.value);
  };
  const filterProducts = data.filter((product) =>
    product.product_type.toLowerCase().includes(value.toLowerCase()));
  return (
    <>
    <Categories data={data} handleClick={handleClick} />
      <div className="products">
        {value === 'Select a product type'
         ? data.map((product) => (
          <div key={product.id}>
            <p className="head">
              {product.name}
              <Btn id={product.id} />
            </p>
            <p>{product.product_type}</p>
            <p>{product.price}</p>
            {/* <p key={product.id}>{product.description}</p> */}
            {/* <p key={product.id}>
              <img src={product.image_link} alt={product.title} />
            </p> */}
          </div>
        ))
        
      </div>
    </>
  );
};

export default HomePage;
