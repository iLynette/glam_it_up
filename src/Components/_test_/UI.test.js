import { BrowserRouter as Router } from "react-router-dom";
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import HomePage from '../HomePage';
import store from '../../redux/configStore';

const HomePageProvider = () => (
    <Provider store={store}>
        <Router>
            <HomePage />
        </Router>
    </Provider>
)

describe('The homepage renders its UI properly', () => {
    it('UI renders correctly', () => {
        const homePage = renderer.create(<HomePageProvider />).toTree()
        expect(homePage).toMatchSnapshot();
    })
})