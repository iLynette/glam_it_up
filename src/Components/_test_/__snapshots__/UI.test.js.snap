// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The homepage renders its UI properly UI renders correctly 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {},
  "rendered": Object {
    "instance": null,
    "nodeType": "component",
    "props": Object {
      "children": <BrowserRouter>
        <HomePage />
      </BrowserRouter>,
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
    "rendered": Object {
      "instance": null,
      "nodeType": "component",
      "props": Object {
        "children": <HomePage />,
      },
      "rendered": Object {
        "instance": null,
        "nodeType": "component",
        "props": Object {
          "basename": undefined,
          "children": <HomePage />,
          "location": Object {
            "hash": "",
            "key": "default",
            "pathname": "/",
            "search": "",
            "state": null,
          },
          "navigationType": "POP",
          "navigator": Object {
            "action": "POP",
            "back": [Function],
            "block": [Function],
            "createHref": [Function],
            "forward": [Function],
            "go": [Function],
            "listen": [Function],
            "location": Object {
              "hash": "",
              "key": "default",
              "pathname": "/",
              "search": "",
              "state": null,
            },
            "push": [Function],
            "replace": [Function],
          },
        },
        "rendered": Object {
          "instance": null,
          "nodeType": "component",
          "props": Object {},
          "rendered": Array [
            Object {
              "instance": null,
              "nodeType": "component",
              "props": Object {
                "data": Array [],
                "handleClick": [Function],
              },
              "rendered": Object {
                "instance": null,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <option
                      value="select a product type"
                    >
                      select a product type
                      
                    </option>,
                    Array [],
                  ],
                  "className": "items",
                  "onChange": [Function],
                  "onClick": [Function],
                  "required": true,
                },
                "rendered": Array [
                  Object {
                    "instance": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "select a product type",
                        "",
                      ],
                      "value": "select a product type",
                    },
                    "rendered": Array [
                      "select a product type",
                    ],
                    "type": "option",
                  },
                  null,
                ],
                "type": "select",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [],
                "className": "products",
              },
              "rendered": Array [],
              "type": "div",
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  "type": [Function],
}
`;

exports[`renders details properly UI renders correctly 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {},
  "rendered": Object {
    "instance": null,
    "nodeType": "component",
    "props": Object {
      "children": <BrowserRouter>
        <DetailsPage />
      </BrowserRouter>,
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
    "rendered": Object {
      "instance": null,
      "nodeType": "component",
      "props": Object {
        "children": <DetailsPage />,
      },
      "rendered": Object {
        "instance": null,
        "nodeType": "component",
        "props": Object {
          "basename": undefined,
          "children": <DetailsPage />,
          "location": Object {
            "hash": "",
            "key": "default",
            "pathname": "/",
            "search": "",
            "state": null,
          },
          "navigationType": "POP",
          "navigator": Object {
            "action": "POP",
            "back": [Function],
            "block": [Function],
            "createHref": [Function],
            "forward": [Function],
            "go": [Function],
            "listen": [Function],
            "location": Object {
              "hash": "",
              "key": "default",
              "pathname": "/",
              "search": "",
              "state": null,
            },
            "push": [Function],
            "replace": [Function],
          },
        },
        "rendered": Object {
          "instance": null,
          "nodeType": "component",
          "props": Object {},
          "rendered": Object {
            "instance": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
              "className": "details",
            },
            "rendered": Array [],
            "type": "div",
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  "type": [Function],
}
`;
